x-pack/plugins/canvas/public/components/arg_form/arg_template_form.tsx

```diff
@@ -5,10 +5,10 @@
  * 2.0.
  */
 
-import React, { useState, useEffect, useCallback, useRef } from 'react';
+import React, { useState, useEffect, useCallback, useRef, memo, ReactPortal } from 'react';
```

Comment:
Is this arg_template_form even used by the datasource at all? do we really need all of these changes? Can it just be the datasource to use the Portal, or is this just setting up for the future?

Reply:
@crob611, UI arguments and datasources are generated by `templateFromReactComponent` function.

https://github.com/elastic/kibana/blob/71bd575a7aa8617e01ba44121051e028e67c883b/x-pack/plugins/canvas/canvas_plugin_src/uis/datasources/essql.js#L117

https://github.com/elastic/kibana/blob/71bd575a7aa8617e01ba44121051e028e67c883b/x-pack/plugins/canvas/canvas_plugin_src/uis/arguments/vis_dimension.tsx#L95

It returns `ReactPortal`.

https://github.com/elastic/kibana/blob/71bd575a7aa8617e01ba44121051e028e67c883b/x-pack/plugins/canvas/public/lib/template_from_react_component.tsx#L80

That's why it is necessary to add support of handling `ReactPortal` in all the places templates are used.